### ISSUE TITLE
Snapshot failure results in frozen VM

### ISSUE THREAD

**Issue by tmuncks** [vmware/open-vm-tools#27]
Whenever a quiesced snapshot is created from VMware, the tools will issue the freeze ioctl just before and the thaw ioctl just after the actual snapshot.

If for some reason the snapshot fails, the thaw ioctl is never issued and we have a frozen VM.

With 9.10.0, a stale -ctk.vmdk from a non-existing snapshot prevented a proper snapshot on the host side - This causes this issue, but I suspect, the problem is more generic in nature.

Would it be possible to add some sort of failsafe mechanism, to make sure we always bail out and unfreeze everything - or, if the tools are informed that the snapshot has failed, just handle it on that event? It seems a crash-consistent backup is preferable to a production server down. It is to me at least.

One way to reproduce (provoke) this behaviour on my end:

1) Create a test VM with no snapshots, with CBT enabled and with open-vm-tools 9.10 installed
2) Create a properly named snapshot CBT file (creating a naming conflict). For "test-app.vmdk", create an empty file named "test-app-000001-ctk.vmdk".
3) take a non-qiuescing snapshot (this validates the conflict and VMware should throw the error; Create VM snapshot test-app the virtual disk is either corrupted or not a supported format)
4) Now take a quiescing snapshot and notice how the system is frozen.

Let be know if I can provide you guys with more info.

**Comment by ravindravmw** [comment]
Could you please collect and send vmware.log from the failed VM to my email id? Thanks!

**Comment by tmuncks** [comment]
Hope the provided log has all the information you need. Please let me know if you need anything else.

**Comment by ukguy** [comment]
We have centos 6.6 VM's which crash during snapshot quiescing due to a problem between the centos 6 kernel and vmware tools. This seems similar to the issue above.

They recommend
Add to file "/etc/vmware-tools/tools.conf" this:
[vmbackup]
enableSyncDriver=false

However, I cannot find out if disabling syncdriver then gives us a non-consistent snapshot.

**Comment by ravindravmw** [comment]
@ukguy Yes, disabling the syncDriver will get you a crash-consistent snapshot. Also, please open a support case with VMware for VMware Tools issues.

**Comment by ukguy** [comment]
My question was 9 months ago but thanks for confirming.
VMware will not support the issue because it's a bug in the centos 6 kernel triggered when trying to quiesce.

**Comment by seisen** [comment]
same here with CentOS 7.1.1503
Linux bt1svlie 3.10.0-229.14.1.el7.x86_64 #1 SMP Tue Sep 15 15:05:51 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
open-vm-tools-9.10.2-4.el7.x86_64

If i take a snapshot: 
 - without mem snap 
 - with quiesce
(other combinaition are working but no "freezing" in log)

/var/log/vmware-vmsvc.log says:
[Oct 11 11:27:03.745] [   debug] [vmsvc] SyncDriver: Freezing using Linux ioctls...
[Oct 11 11:27:03.758] [   debug] [vmsvc] SyncDriver: successfully froze '/'.

And this is the end of the log, disk are frozen.

The workaround works:
[vmbackup]
enableSyncDriver=false

**Comment by ravindravmw** [comment]
@seisen Enabling logs to go to filesystem when you intend to do filesystem quiescing could lead to vmtoolsd deadlock in version 9.x and earlier. We fixed it in version 10.x and later. I see that you are using open-vm-tools 9.10.2. Why logs are being generated in /var/log/vmware-vmsvc.log? Could you please share your tools.conf? Could remove tools.conf file or comment out all the lines in it with '#' char and retry?

**Comment by ravindravmw** [comment]
tools.conf can be found under /etc/vmware-tools/tools.conf. The logs you have shown are debug logs. For debug logs for quiesce snapshot, it better to use 'vmx' as the handler, e.g. following configuration will redirect the logs to vmware.log on the ESXi host and will work just fine even when guest filesystem has been quiesced.

# start
[logging]
vmtoolsd.level=debug
vmtoolsd.handler=vmx
vmsvc.level=debug
vmsvc.handler=vmx
vmbackup.level=debug
vmbackup.handler=vmx
# end

NOTE: The configuration above will generate lots of logs in vmware.log, therefore it is recommended only for troubleshooting purpose, but not as general configuration.

If you are seeing quiescing issues even after removing/commenting tools.conf file, please apply the example configuration above, retry the operation and send me a private email with the vmware.log from the failed VM.

You might want to read KB https://kb.vmware.com/kb/1007873 to learn more about logging configuration.

**Comment by seisen** [comment]
Ok, there is a dead lock with quiescing + writing log file.
I decreased the  log level and there is no freeze anymore.
I was using debug to understand failed OS Guest Customization, which is quite tricky
